lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_object_lookup.rb

```diff
@@ -0,0 +1,60 @@
+module MiqAeEngine
+  module MiqAeObjectLookup
+    def initialize_obj_entries
+      @lookup_table = []
+    end
+
+    def add_obj_entry(fqns, klass, instance, object)
+      parts = fqns.split('/')
+      domain = parts.shift
+      namespace = parts.join('/')
+
+      @lookup_table << {:domain    => domain.downcase,
+                        :namespace => namespace.downcase,
+                        :klass     => klass.downcase,
+                        :instance  => instance.downcase,
+                        :object    => object}
+    end
+
+    def find_obj_entry(path)
+      query = path[0] == '/' ? parse_obj_path(path[1..-1]) : parse_obj_path(path)
+
+      entries = @lookup_table.select { |entry| entry[:klass] == query['class_name'] }
+      ientries = entries.select { |entry| instance_match?(query['instance'], entry) }
```

Comment:
What happens if the entries are not found?

Reply:
@Fryguy The substitution call handles it we just return a nil here.
https://github.com/ManageIQ/manageiq-automation_engine/blob/master/lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_object.rb#L399